realm/realm-library/src/androidTest/java/io/realm/RealmAsyncQueryTests.java

```diff
@@ -279,6 +280,117 @@ public void execute(Realm realm) {
         }, transactionCallback);
     }
 
+    // Test case for https://github.com/realm/realm-java/issues/1893
+    // Ensure that onSuccess might not be called with the right Realm version for async transaction.
+    @Test
+    @RunTestInLooperThread
+    public void executeTransactionAsync_asyncQuery() {
+        final Realm realm = looperThread.realm;
+        RealmResults<AllTypes> results = realm.where(AllTypes.class).findAllAsync();
+        looperThread.keepStrongReference.add(results);
+        realm.executeTransactionAsync(new Realm.Transaction() {
+            @Override
+            public void execute(Realm realm) {
+                realm.createObject(AllTypes.class);
+            }
+        }, new Realm.Transaction.OnSuccess() {
+            @Override
+            public void onSuccess() {
+                assertEquals(realm.where(AllTypes.class).count(), 1);
+                looperThread.testComplete();
+            }
+        }, new Realm.Transaction.OnError() {
+            @Override
+            public void onError(Throwable error) {
+                fail();
+            }
+        });
+    }
+
+    // Test case for https://github.com/realm/realm-java/issues/1893
+    // Ensure that Realm.close() logs a warning if you close a Realm while an async transaction is pending.
+    @Test
+    @RunTestInLooperThread
+    public void executeTransactionAsync_closeWithPendingTransaction() {
+        final CountDownLatch queryLatch = new CountDownLatch(1);
+        final CountDownLatch closeLatch = new CountDownLatch(1);
+
+        Realm realm = looperThread.realm;
+        Realm.asyncTaskExecutor.submit(new Callable<Long>() {
+            @Override
+            public Long call() throws Exception {
+                TestHelper.awaitOrFail(queryLatch);
+                return 0L;
+            }
+        });
+        realm.executeTransactionAsync(new Realm.Transaction() {
+            @Override
+            public void execute(Realm realm) {
+                realm.close();
+                closeLatch.countDown();
+            }
+        });
+
+        final TestHelper.TestLogger testLogger = new TestHelper.TestLogger();
+        RealmLog.add(testLogger);
+        realm.close();
+        RealmLog.remove(testLogger);
+        queryLatch.countDown();
+
+        String[] logs = testLogger.message.split("will ");
+        assertEquals("be closed with pending async transactions or callbacks.", logs[1]);
+
+        TestHelper.awaitOrFail(closeLatch);
+        looperThread.testComplete();
```

Comment:
If this is the last statement it is usually an indication you can replace `@RunTestInLooperThread` with the standard `@UiThreadTest`